app.js

```diff
@@ -0,0 +1,221 @@
+const token = require("./token.json");
+const Discord = require("discord.js");
+var fs = require("fs");
+var _ = require("lodash");
+var cronstrue = require("cronstrue");
+const schedule = require("node-schedule");
+const { time } = require("console");
+var data = fs.readFileSync("data.json");
+
+// parsing data.json
+data = JSON.parse(data);
+
+// creating client
+const client = new Discord.Client({
+  intents: [Discord.Intents.FLAGS.GUILDS, Discord.Intents.FLAGS.GUILD_MESSAGES],
+});
+
+// categories
+categories = [
+  "WILDCARD",
+  "LIVING BETTER",
+  "BUSINESS TECH",
+  "HISTORY CULTURE",
+  "SCIENCE NATURE",
+];
+
+// command message
+var command =
+  "1. **For getting a random article**: ```get article [category]``` " +
+  "\n" +
+  "ex: `get article wildcard`, this will fetch a random article form wildcard category" +
+  "\n" +
+  "\n" +
+  "Tip: `get article`, this will give you the list of categories available" +
+  "\n" +
+  "\n" +
+  "Tip: `get article time`, this will give you the time at which you get the daily article" +
+  "\n" +
+  "\n" +
+  "2. **For setting time of daily message**:" +
+  "\n" +
+  "\n" +
+  "> 1. for days(in integers from 0-6): ```set article days [startDay] [endDay]```" +
+  "\n" +
+  "ex: `set article days 0 6`, this will set the days as SUN-SAT" +
+  "\n" +
+  "\n" +
+  "> 2. for time(in 24hr format): ```set article time [hour]:[mins]```" +
+  "\n" +
+  "ex: `set article time hour 14:20`, this will give the daily article at 2:20 pm";
+
+// embed command message
+const exampleEmbed = new Discord.MessageEmbed()
+  .setColor("#FF3F3F")
+  .setTitle("readsomethinggreat")
+  .setURL("https://www.readsomethinggreat.com/")
+  .setDescription(
+    "A bot that gives a random article from readsomethinggreat.com"
+  )
+  .setThumbnail("https://i.imgur.com/vugPtoT.png")
+  .addFields(
+    {
+      name: "Categories",
+      value:
+        "Wildcard, Living Better, Business & Tech, History & Culture, Science & Nature",
+    },
+    { name: "\u200B", value: "\u200B" },
+    {
+      name: "Commands",
+      value: command,
+      inline: true,
+    }
+  )
+  .setTimestamp()
+  .setFooter("Happy Reading", "https://i.imgur.com/vugPtoT.png");
+
+const rule = new schedule.RecurrenceRule();
+rule.dayOfWeek = [0, new schedule.Range(0, 6)];
+rule.hour = 10;
+rule.minute = 00;
+var startDay = 0;
+var endDay = 6;
+
+//creating a empty cron expression
+var cronExpression = `${rule.minute} ${rule.hour} * * ${startDay}-${endDay}`;
+
+var dailyUpdatesChannel = null;
+
+// fetching random article
+function fetchRandomArticle(category) {
+  lengthOfArticle = data[category].length;
```

Comment:
have better names of variables, the name of the variable should describe what the variable contains.

use `numberOfArticles` instead of `lengthOfArticle`

Reply:
yeah ok